### PR TITLE
Fixing issues reported by binary analysis - Hardening checks

### PR DESCRIPTION
**Pull request by ulvii**
- Adding compiler options `-D_FORTIFY_SOURCE` and `-fstack-protector`
- Adding linker option  `-Wl,-z,now` for Linux and `-Wl,-bind_at_load` for Max
- Fixing the issue with detecting the platform

### CODE REVIEW

**Comment by ulvii** [comment]
Failed because pdo_sqlsrv.so was loaded before pdo.so after the changes

**Comment by ulvii** [comment]
Adding optimizations flag decreases the coverage, because some lines are not hit after the optimization. Turning on optimization flags makes the compiler attempt to improve the performance and/or code size at the expense of compilation time and possibly the ability to debug the program. This flag should be turned off when debugging.

File: source/pdo_sqlsrv/config.m4 [hunk 1]
```diff
@@ -52,8 +52,14 @@ if test "$PHP_PDO_SQLSRV" != "no"; then
   fi
     AC_MSG_RESULT($pdo_sqlsrv_inc_path)
     
-           
+  HOST_OS_ARCH=`uname`         
+  if test "${HOST_OS_ARCH}" = "Darwin"; then
+      MACOSX_DEPLOYMENT_TARGET=`sw_vers -productVersion` 
+  fi
```
Comment (v-kaywon) [inline source/pdo_sqlsrv/config.m4]: why does this guy need to be moved up?

File: source/sqlsrv/config.w32 [hunk 2]
```diff
@@ -3,7 +3,7 @@
 //
 // Contents: JScript build configuration used by buildconf.bat
 //
-// Microsoft Drivers 4.3 for PHP for SQL Server
+// Microsoft Drivers 4.2 for PHP for SQL Server
```
Comment (v-kaywon) [inline source/sqlsrv/config.w32]: shouldn't it be 4.3??

File: source/pdo_sqlsrv/config.m4 (hunk 1 above)
Comment (ulvii) [inline source/pdo_sqlsrv/config.m4]: It should be inside the if block.
Also, the previous way of checking the OS was wrong.

File: source/sqlsrv/config.w32 (hunk 2 above)
Comment (ulvii) [inline source/sqlsrv/config.w32]: it should :)

**Pull request merged**